src/common/zone.c

```diff
@@ -125,14 +128,29 @@ void Z_Free(void *ptr)
     Z_CountFree(z);
 
     if (z->tag != TAG_STATIC) {
-        z->prev->next = z->next;
-        z->next->prev = z->prev;
+        List_Remove(&z->entry);
         z->magic = 0xdead;
         z->tag = TAG_FREE;
         free(z);
     }
 }
 
+/*
+========================
+Z_Freep
+========================
+*/
+void Z_Freep(void *ptr)
```

Comment:
Suggest changing the parameter type to `void **p` to make it clearer what `Z_Freep` is doing. And fixing a couple of warnings about type mismatches where this function is used to deallocate arrays of pointers.

Reply:
Done.
The clarity of what it does is a good point, but I suspect the additional warnings was the reason this is `void*` instead of `void**`.
They can be avoided using casts (which I added), but I think having to use one for every call to `Z_Freep()` isn't strictly better than just having take a `void*`. (Dunno, maybe something with macros that could be done?)

Reply:
Thanks!
I was only getting 2 warnings when I tried it though, using MSVC 2019. Which compiler are you using? Just curious.

Reply:
gcc warns extensively. Version 11, in my case.